clevertap-core/src/main/java/com/clevertap/android/sdk/pushnotification/PushProviders.java

```diff
@@ -605,24 +601,6 @@ private List<CTPushProvider> createProviders() {
             providers.add(pushProvider);
         }
 
-        for (PushType pushType : allDisabledPushTypes) {
-            // only for XPS, if for disabled push cached token already exists then unregister xiaomi push
-            // for case like user enables xps on all devices first then in next app version disables on all devices
-
-            if (pushType == PushType.XPS) {
-                String cachedTokenXps = getCachedToken(PushType.XPS);
-                if (!TextUtils.isEmpty(cachedTokenXps)) {
-                    CTPushProvider pushProvider = getCTPushProviderFromPushType(pushType, false);
-
-                    if (pushProvider instanceof UnregistrableCTPushProvider) {
-                        ((UnregistrableCTPushProvider) pushProvider).unregisterPush(context);
```

Comment:
@CTLalit We can delete `UnregistrableCTPushProvider` file

Reply:
done